packages/next/src/build/utils.ts

```diff
@@ -1554,6 +1554,15 @@ export function detectConflictingPaths(
   >()
 
   const dynamicSsgPages = [...ssgPages].filter((page) => isDynamicRoute(page))
+  const additionalSsgPathsByPath: {[key: string]: {[key: string]: string}} = {};
+
+  additionalSsgPaths.forEach((paths, pathsPage)=>{
+    additionalSsgPathsByPath[pathsPage] ||= {};
+    paths.forEach((curPath)=>{
+      const currentPath = curPath.toLowerCase();
+      additionalSsgPathsByPath[pathsPage][currentPath] = curPath;
+    });
+  });
```

Comment:
instead of a `[].find(x => x.find())` (see comment below),  we first create a lookup-table of routes that contain full paths. This lets us get O(1) lookup speeds inside the surrounding `dynamicSsgPages.find()` search, dramatically increasing the performance of this function when sites have hundreds of thousands of paths.

Reply:
You probably need to fix some lint errors here.

Reply:
@shuding should be fixed in https://github.com/vercel/next.js/pull/46080/commits/3582b66e180f2484b7efefc1d5ff127192c7a5b0